java/ql/src/Security/CWE/CWE-502/UnsafeDeserialization.ql

```diff
@@ -21,6 +21,39 @@ class UnsafeDeserializationConfig extends TaintTracking::Configuration {
   override predicate isSource(DataFlow::Node source) { source instanceof RemoteFlowSource }
 
   override predicate isSink(DataFlow::Node sink) { sink instanceof UnsafeDeserializationSink }
+
+  override predicate isAdditionalTaintStep(DataFlow::Node prod, DataFlow::Node succ) {
+    exists(ClassInstanceExpr cie |
+      cie.getConstructor().getDeclaringType() instanceof JsonReader and
+      cie.getArgument(0) = prod.asExpr() and
+      cie = succ.asExpr() and
+      not exists(SafeJsonIo sji | sji.hasFlowToExpr(cie.getArgument(1)))
+    )
+    or
+    exists(ClassInstanceExpr cie |
```

Comment:
Once the `not exists(SafeJsonIo sji | sji.hasFlowToExpr(cie.getArgument(1)))` has been moved to `isSanitizer`, you can factor this like
```
exists(ClassInstanceExpr cie |
      cie.getArgument(0) = prod.asExpr() and
      cie = succ.asExpr() and
(
      cie.getConstructor().getDeclaringType() instanceof YamlReader or
      cie.getConstructor().getDeclaringType() instanceof JsonReader or
      ...
)
```

Reply:
Thanks for the suggestion.